go/vt/tabletmanager/state_change.go

```diff
@@ -293,6 +293,27 @@ func (agent *ActionAgent) changeCallback(ctx context.Context, oldTablet, newTabl
 		}
 	}
 
+	// See if we need to enable or disable semi-sync replication.
+	// Note we never enable the master side, as that would have
+	// been done by either the reparent process, or an external
+	// tool (like Orchestrator).
+	if *enableSemiSync {
```

Comment:
I am looking at this right now. I think I found one bug at least: after a backup restore, it seems we use the wrong type (RESTORE) to decide if we need to enable semi-sync.

So I think I will change the logic of the checker part: as you suggest, always call enableSemiSync after a state change. But then, change the code: if replication is running, and the semi-sync status is different than the variable, restart replication.